internal/cli/install.go

```diff
@@ -132,15 +134,77 @@ func (c *InstallCommand) Run(args []string) int {
 		)
 		return 1
 	}
+
 	if tokenResp != nil {
 		log.Debug("token received, setting on context")
 		contextConfig.Server.RequireAuth = true
 		contextConfig.Server.AuthToken = tokenResp.Token
+	} else {
+		// try default context in case server was started again from install
+		defaultCtx, err := c.contextStorage.Default()
+		if err != nil {
```

Comment:
We should never hit this. We overwrite the existing Default context -- or create a new one if it's been deleted -- here https://github.com/hashicorp/waypoint/blob/cfba8f4e229c46d57ddc8f9760f0567b71878f25/internal/serverinstall/docker.go#L81.
Should we move the Docker context creation down, after the `if len(containers) > 0 {` block, and just return a `nil` for an existing install, and then move the context check to line 96 here? Like:
```
contextConfig = result.Context
if contextConfig == nil {
  contextConfig, err = c.contextStorage.Default()
  if err != nil {
    ...
  }
}
```

Reply:
Line 81 in `serverinstall/docker` will create it in a struct, but it doesn't save it through `contextStorage` on disk, so no default has been created or stored yet. The storage part and setting the default ends up happening here in install after the bootstrap token step right after this on 208.

We'll want this here after the platform install step, since the installer attempts to bootstrap the server after the platform install. The bootstrap step will tell us whether or not the server has been set up already, so here is where we can check to see if the context used to invoke the install command to restart the server has the right auth token. Otherwise we'd be pulling in all of that bootstrap stuff from here into the docker serverinstall platform which it shouldn't be handling.

I think we should try to keep the contextStorage operations and bootstrapping platform agonstic inside this installation function here and just let the serverinstall return the `InstallResults` struct for us to handle.

Reply:
If I `waypoint context delete` and `docker stop waypoint-server` I make it past this error catch and end up erroring out with
```
❌ Configuring server...
! Error setting the advertise address: Authorization token is not supplied
```
Which is technically correct; the context that was created for me during the re-install of the old server does not have an authorization token in it, since I deleted the old context + auth token. So this current behavior is fine, I just wanted to make sure the error catch here will catch what we expect it to.

Reply:
@krantzinator - Ah, I see what you're saying. Yeah that's a weird case :thinking: Agreed, I think we can provide a better error message in this case.